plugins/modules/proxmox_snap_info.py

```diff
@@ -0,0 +1,170 @@
+  register: snaplist
+
+'''
+
+RETURN = r'''#'''
+
+from ansible_collections.community.general.plugins.module_utils.proxmox import (
+    ansible_to_proxmox_bool, proxmox_auth_argument_spec, ProxmoxAnsible)
+import traceback
+from ansible.module_utils.common.text.converters import to_native
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+import time
+
+try:
+    from ansible_collections.community.general.plugins.modules.proxmox_snap import (
+        ProxmoxSnapAnsible)
+
+    HAS_PROXMOX_SNAP = True
+    PROXMOX_SNAP_IMPORT_ERROR = None
+except ImportError:
+    HAS_PROXMOX_SNAP = False
+    PROXMOX_SNAP_IMPORT_ERROR = traceback.format_exc()
+else:
+    HAS_PROXMOX_SNAP = True
+
+
+def get_proxmox_snapshotlist(module, hostname, vmid):
+    proxmox = ProxmoxSnapAnsible(module)
+    if not vmid and hostname:
+        vmid = proxmox.get_vmid(hostname)
+    elif not vmid:
+        module.exit_json(changed=False, msg="Vmid could not be fetched")
+
+    vm = proxmox.get_vm(vmid)
+    snapshotlist = proxmox.snapshot(vm, vmid).get()
+    if not snapshotlist:
+        module.exit_json(changed=False, msg="Snapshots could not be fetched")
+
+    return snapshotlist
+
+
+def main():
+    module_args = proxmox_auth_argument_spec()
+    snap_args = dict(
+        vmid=dict(required=False),
+        hostname=dict(),
+        timeout=dict(type='int', default=30),
+        snapname=dict(type='str', required=False),
+        getsnapshots=dict(type='bool', default=False),
+        older_than=dict(type='int', default=0)
+    )
+    module_args.update(snap_args)
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+
+    if not HAS_PROXMOX_SNAP:
+        # Needs: from ansible.module_utils.basic import missing_required_lib
+        module.fail_json(
+            msg=missing_required_lib('proxmox_snap'),
+            exception=PROXMOX_SNAP_IMPORT_ERROR)
+
+    vmid = module.params['vmid']
+    hostname = module.params['hostname']
+    snapname = module.params['snapname']
+    timeout = module.params['timeout']
+    getsnapshots = module.params['getsnapshots']
+    older_than = module.params['older_than']
+
+    if getsnapshots:
+        snapshotlist = get_proxmox_snapshotlist(module, hostname, vmid)
+        oldsnapshotlist = []
+
+        for s in snapshotlist:
+            if s["name"] == "current":
+                continue
+            if snapname:
+                if not s["name"][0:len(snapname)] == snapname:
+                    continue
+            if ((time.time() - s["snaptime"]) / 60 / 60 / 24) > older_than:
+                oldsnapshotlist.append(s["name"])
+
+        snapshotdict = {
+            "changed": False,
+            "results": oldsnapshotlist,
+            "older_than": older_than}
+
+        module.exit_json(**snapshotdict)
+    else:
+        result = dict(changed=False)
+
+        result['msg'] = 'No query requested: try "getsnapshot"'
```

Comment:
I don't think this behavior makes sense. Why should the module do nothing when `getsnapshots` is not set to `true`? It sounds to me like this parameter should be removed.